packages/channels-app/src/screens/add.channel.screen.tsx

```diff
@@ -71,20 +68,30 @@ function AddChannelScreen() {
     },
     onSuccess: () => {
       hideBackdrop()
-      navigate(Routes.MANAGE_CHANNELS)
+      window.location.href = `/#${Routes.MANAGE_CHANNELS}`
     },
     onError: (error: any) => {
+      const err = error?.response?.data ?? 'An unexpected error occurred'
       console.error(error)
       hideBackdrop()
+      showAlert('Error creating a new channel. ' + err, 'Error', 'error')
+    }
+  })
+
+  const handleAddChannel = () => {
+    const numOfPrimaryRoutes = channel.routes?.filter(
+      route => !!route.primary
+    ).length
+
+    if (numOfPrimaryRoutes !== 1) {
       showAlert(
-        'Error creating a new channel. ' + error?.response?.data,
+        'Channel must have exactly only 1 primary route.',
```

Comment:
Shouldn't we handle this on the add route modal?

Reply:
It's only at this component whereby we have all the information that we need about all of the channels and their respective routes.

At the deeper nested components, we only have access to the 1 individual route that we're modifying and/or adding.